File: account/src/main/java/serp/project/account/core/domain/dto/request/CreateUserForOrgRequest.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.account.core.domain.dto.request;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import serp.project.account.core.domain.enums.UserType;

import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public class CreateUserForOrgRequest {
    @NotBlank
    private String firstName;
    @NotBlank
    private String lastName;
    @NotBlank
    private String email;
    @NotBlank
    private String password;
    @Builder.Default
    private UserType userType = UserType.EMPLOYEE;

    private List<@NotNull Long> roleIds;
}

File: api_gateway/src/core/domain/dto/request/account/create_user_for_org_request.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package request

type CreateUserForOrgRequest struct {
	FirstName string   `json:"firstName"`
	LastName  string   `json:"lastName"`
	Email     string   `json:"email"`
	Password  string   `json:"password"`
	UserType  *string  `json:"userType"`
	RoleIds   []*int64 `json:"roleIds"`
}
